src/Platforms/Gcc/UtestPlatform.cpp

```diff
@@ -111,7 +115,7 @@ static void GccPlatformSpecificRunTestInASeperateProcess(UtestShell* shell, Test
                 result->addFailure(TestFailure(shell, "Stopped in separate process - continuing"));
                 kill(w, SIGCONT);
             }
-        } while (!WIFEXITED(status) && !WIFSIGNALED(status));
+        } while ((w == syscallError) || (!WIFEXITED(status) && !WIFSIGNALED(status)));
```

Comment:
This prevents reading uninitialized `status` if `continue` from line 100 happens (I've also updated the test to validate that he loop does not exit after the 1st iteration as it did before randomly)

Reply:
Confusing. Could you remove the continue by reverting the if on line 100 and changing the next if to an else if. That ought to make the logic a bit clearer. Having continue in a do-while loop feels bad.

Perhaps also extract the status logic of creating the right failure message in a separate function.

Reply:
After refactoring, it became evident to me that `WIFSTOPPED()` case looks weird:
1. it is not listed in while() -> seems logical as we resume the process via `SIGCONT`;
2. but there is a `TestFailure` added in such case - why? if we resume the process and wait, it means it did not fail...
As this goes beyond my understanding and debugging capabilities, I propose to create an issue for this scenario. Someone may address it if they face it.

Reply:
Done (I kept `waitpid()` failure in the same function to reduce code movement - it seems to me to be ok, but I can try to move it as well)